File: Java-BackEnd/JansiSnack_BackEnd/Api-commons/src/main/java/tech/wetech/api/model/Category.java
package tech.wetech.api.model;

import jakarta.persistence.*;

/**
 * @author Jansonci
 */
@Entity
@Table(name = "categories")
public class Category {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(length = 100, nullable = false, unique = true)
  private String name;

  @Column()
  private String imageUrl;

  public Category() {
  }

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public String getImageUrl() {
    return imageUrl;
  }

  public void setImageUrl(String imageUrl) {
    this.imageUrl = imageUrl;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public void addDessert(Dessert dessert) {
    dessert.setCategory(this.name);
  }
}

File: Kotlin-Android/JansiSnack_Front/app/src/main/java/com/example/jetsnack/model/Category.kt
package com.example.jetsnack.model

import androidx.room.Entity

@Entity(tableName = "categories")
class Category(
    id: Long,
    imageUrl: String,
    name: String,
    // 你可以在这里添加 Category 特有的属性
) : Kind(id, imageUrl, name) // 正确地调用 Kind 的构造器